### PR TITLE
compattests: Add tests for current v1 TPM metadata format

### PR DESCRIPTION
**Pull request by chrisccoulson**

### CODE REVIEW

File: internal/compattest/v0_test.go [hunk 1]
```diff
@@ -122,7 +122,7 @@ func (s *compatTestV0Suite) TestUnsealAfterLock(c *C) {
 	// Test unsealing a v0 file from a newer initramfs using the fence-style locking - this just makes
 	// the PCR values invalid so there's no reason this shouldn't work or require a compatibility test,
 	// but keep this here just to make sure.
-	c.Assert(secboot.BlockPCRProtectionPolicies(s.TPM, []int{12}), IsNil)
 	s.replayPCRSequenceFromFile(c, s.absPath("pcrSequence.1"))
+	c.Assert(secboot.BlockPCRProtectionPolicies(s.TPM, []int{12}), IsNil)
```
Comment (pedronis) [inline internal/compattest/v0_test.go:126]: why this change?

**Review by pedronis — COMMENTED** [review]
looks good but a question

File: internal/compattest/v0_test.go (hunk 1 above)
Comment (chrisccoulson) [reply internal/compattest/v0_test.go:126]: Although the end result is the same (albeit with different PCR values afterwards), the original order isn't what happens on a real system and this just corrects that.

**Review by pedronis — APPROVED** [review]
thanks
